src/models/operation-trait.ts

```diff
@@ -1,17 +1,11 @@
 import type { BaseModel } from './base';
-import type { ChannelsInterface } from './channels';
-import type { OperationAction } from './operation';
 import type { SecurityRequirements } from './security-requirements';
 import type { BindingsMixinInterface, DescriptionMixinInterface, ExtensionsMixinInterface, ExternalDocumentationMixinInterface, TagsMixinInterface } from './mixins';
 
 export interface OperationTraitInterface extends BaseModel, BindingsMixinInterface, DescriptionMixinInterface, ExtensionsMixinInterface, ExternalDocumentationMixinInterface, TagsMixinInterface {
-  id(): string | undefined;
   hasId(): boolean;
-  action(): OperationAction | undefined;
-  isSend(): boolean;
-  isReceive(): boolean;
+  id(): string | undefined;
   hasSummary(): boolean;
   summary(): string | undefined;
   security(): SecurityRequirements[];
-  channels(): ChannelsInterface;
```

Comment:
Is there a reason why channels arent part of traits?

Reply:
https://github.com/asyncapi/spec/pull/860 we had PR for that in spec. There is also a link to the issue for it :)

Reply:
Maybe in the future we add such a information in ParserJS to check in which channels (where operations are assigned) given trait is applied, but not now :)

Reply:
You can even create PR in `parser-api` for such a method(s) :)